src/NuGet.Clients/PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -862,7 +863,14 @@ private void SuppressDisclaimerChecked(object sender, RoutedEventArgs e)
             RegistrySettingUtility.SetBooleanSetting(Constants.SuppressUIDisclaimerRegistryName, true);
         }
 
-        public void ExecuteAction(UserAction action, Action<NuGetUI> setOptions)
+        /// <summary>
+        /// This method is called after user clicks a button to execute an action, e.g. install a package.
+        /// </summary>
+        /// <param name="performAction">A function that returns the task that is performing the action 
+        /// thru UIActionEngine.</param>
+        /// <param name="setOptions">A method that is called to set the action options, 
+        /// such as dependency behavior.</param>
+        public void ExecuteAction(Func<Task> performAction, Action<NuGetUI> setOptions)
```

Comment:
Changed this method so that it takes perfomAction function, so it can be called both for execution a single user action (install/uninstall) and for multi-package update.